### ISSUE TITLE
deposited artio fields have slightly different answers under yt-4.0

### ISSUE THREAD

**Issue by ngoldbaum** [yt-project/yt#2195]
This is causing answer test failures on #2172.

**Code for reproduction**

```python
import yt

ds = yt.load('sizmbhloz-clref04SNth-rs9_a0.9011/sizmbhloz-clref04SNth-rs9_a0.9011.art')

sph = ds.sphere('max', (0.1, 'unitary'))

count = sph['deposit', 'all_count']

print(count.sum())
```

**Actual outcome**

On master this prints `201313.0 dimensionless` whereas on yt-4.0 it prints `201134.0 dimensionless`.

**Comment by ngoldbaum** [comment]
Here's a slightly modified version of the script in the description:

```
import yt
import numpy as np

ds = yt.load('sizmbhloz-clref04SNth-rs9_a0.9011/sizmbhloz-clref04SNth-rs9_a0.9011.art')

sph = ds.sphere('max', (0.1, 'unitary'))

count = sph['deposit', 'all_count']

counts_master = yt.YTArray.from_hdf5('counts_master.h5')

print(count.sum())

wh = np.where(count != counts_master)

print(count[wh])

print(counts_master[wh])
```

This uses a file I've uploaded here: http://use.yt/upload/17eccce3

On the yt-4 branch this prints:

```
[0. 0. 0. 0. 0. 0. 0. 0. 1. 0. 0. 0. 0. 0. 0. 0. 1. 0. 0. 0. 0. 0. 0. 0.
 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 1. 0. 0. 1. 0. 0. 0. 0. 1. 0. 0.
 0. 1. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 2. 0. 0. 0. 0. 0. 0. 1. 0. 0. 0. 0.
 0. 1. 0. 0. 0. 0. 0. 0. 0. 1. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 1. 0. 0. 0.
 0. 0. 0. 1. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0.
 0. 0. 0. 0. 0. 1. 0. 0. 0. 0. 0. 0. 1. 0. 0. 0. 0. 0. 0. 0. 0. 1. 1. 1.
 0. 0. 1. 0. 0. 0. 1. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0.
 1. 0. 0. 0. 0. 0. 0.] dimensionless
[1. 1. 2. 1. 1. 1. 1. 1. 2. 1. 1. 1. 1. 1. 1. 1. 3. 1. 1. 1. 1. 1. 1. 1.
 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 2. 1. 1. 2. 1. 1. 1. 2. 2. 1. 1.
 1. 2. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 3. 1. 1. 1. 1. 1. 1. 2. 1. 1. 1. 1.
 1. 2. 1. 1. 1. 1. 1. 1. 1. 2. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 2. 1. 1. 1.
 1. 1. 1. 2. 1. 1. 1. 1. 1. 1. 1. 1. 1. 2. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1.
 1. 1. 1. 1. 1. 2. 1. 1. 1. 1. 1. 1. 2. 1. 1. 1. 1. 1. 1. 1. 1. 2. 2. 2.
 1. 1. 2. 1. 1. 1. 2. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1.
 2. 1. 1. 1. 1. 1. 1.] dimensionless
```

So it looks like in all cases this is due to the deposit operation missing a single particle.

**Comment by MatthewTurk** [comment]
This is interesting.  It's probably worth checking that the same number of particles is selected by the sphere in both cases.

**Comment by MatthewTurk** [comment]
I have tracked this down to a single change in `selection_routines.pyx`.  In `master`, under the `OctreeSubsetSelector.select_point` we return 1 always; this means that if the oct is selected, everything within that oct is also selected.  In `yt-4.0`, this dispatches to the `base_selector` and asks it.

So, this results in a difference when doing a `count` field (i.e., assigning all particles to their oct cells) when the oct *cell* would be selected but the *particle* would not.  The particular case we're seeing is when a sphere hits an oct cell, and the surface of the sphere separates some of the particles.  What is the value we should have for the deposited "count" of particles in an oct?

My assertion is that the new behavior could be described as correct if we are thinking about the process as a "meshing" of particles.  But, if we think about "deposition" fields as somehow being inherent and unchanging based on the selection operation, then the old behavior would be correct.

I've talked myself into circles and I'm not sure which makes more sense, but it is also almost 5.
